doc/sphinx/proofs/writing-proofs/rewriting.rst

```diff
@@ -344,85 +343,94 @@ Rewriting with definitional equality
+      | hnf
+      | simpl {? @delta_flag } {? {| @reference_occs | @pattern_occs } }
+      | cbv {? @strategy_flag }
+      | cbn {? @strategy_flag }
+      | lazy {? @strategy_flag }
+      | compute {? @delta_flag }
+      | vm_compute {? {| @reference_occs | @pattern_occs } }
+      | native_compute {? {| @reference_occs | @pattern_occs } }
+      | unfold {+, @reference_occs }
+      | fold {+ @one_term }
+      | pattern {+, @pattern_occs }
+      | @ident
+      delta_flag ::= {? - } [ {+ @reference } ]
+      strategy_flag ::= {+ @red_flag }
+      | @delta_flag
+      red_flag ::= beta
+      | iota
+      | match
+      | fix
+      | cofix
+      | zeta
+      | delta {? @delta_flag }
+      reference_occs ::= @reference {? at @occs_nums }
+      pattern_occs ::= @one_term {? at @occs_nums }
+
+   Normalize the goal as specified by the :n:`@strategy_flag`\s. The flags
+   include:
+
+   `beta`
+     :term:`beta-reduction` of functional application
+
+   `delta`
+     :term:`delta-reduction`: unfolding of transparent constants, see :ref:` controlling-the-reduction-strategies`
+
+   :n:`delta {? - } [ {+ @reference } ]`
+     without the `-`, limits unfolding to the listed constants.  If the `-` is present,
+     unfolding is applied to all constants that are not listed.
+     Notice that the ``delta`` flag doesn't apply to variables bound by a let-in
+     construction inside the :n:`@term` itself (use the ``zeta`` flag to unfold these).
+     Opaque constants are never unfolded
+     (see :ref:` controlling-the-reduction-strategies`).
+
+   `iota`
+     :term:`iota-reduction` of pattern matching (`match`) over a constructed term and unfolding
+     of :g:`fix` and :g:`cofix` expressions.  Shorthand for `match fix cofix`
+
+   `zeta`
+      :term:`zeta-reduction`: contraction of local definitions
+
+   If :n:`@strategy_flag` is not specified, the conversions `beta delta iota zeta` are used.
+   If :n:`strategy_flag` is a :n:`red_flag`, then only the named conversions are used.
+   If :n:`@strategy_flag` is a :n:`@delta_flag`, then that conversion is applied
+   in addition to `beta delta iota zeta`.
+
+   .. todo PR: following wording needs some work
+
+   Normalization is done by first evaluating the
    head of the expression into a *weak-head* normal form, i.e. until the
-   evaluation is blocked by a variable (or an opaque constant, or an
-   axiom), as e.g. in :g:`x u1 ... un` , or :g:`match x with ... end`, or
-   :g:`(fix f x {struct x} := ...) x`, or is a constructed form (a
+   evaluation is blocked by a variable, an opaque constant or an
+   axiom, such as in :g:`x u1 … un` , or :g:`match x with … end`, or
+   :g:`(fix f x {struct x} := …) x`, or is a constructed form (a
    :math:`\lambda`-expression, a constructor, a cofixpoint, an inductive type, a
-   product type, a sort), or is a redex that the flags prevent to reduce. Once a
+   product type, a sort), or a redex that the flags prevent to reduce. Once a
```

Comment:
So many possible revisions.  Did you get all of them :-) ?

redex = reducible expression, i.e. not normalized?  That deserves a definition somewhere.  39 uses in the doc.  Not the same as "beta-redex", right?

Reply:
> redex = reducible expression, i.e. not normalized? That deserves a definition somewhere. 39 uses in the doc. Not the same as "beta-redex", right?

beta-redex is more specific: it is a term of the form `(fun x => t) t'` (that can thus be reduced by applying a beta-reduction step.
I guess redex generalizes this as any form that can be reduced by any of the reduction steps. Note that reducible expressions can be reducible because they *contain* redexes rather than being redexes themselves.

Reply:
Why use the primed variable `t'`?  What is that mean to suggest about its relationship to `t`?  AFAIK there is no particular relationship.

Reply:
No particular relationship indeed. The reason for the prime, is that it is easier to type than subscripts :wink: